pymantic/primitives.py

```diff
@@ -607,12 +613,17 @@ def resolve(self, curie):
         "http://www.w3.org/2000/01/rdf-schema#label")"""
         return parse_curie(curie, self)
 
-    def shrink(self, iri):
+    def shrink(self, iri, base=None):
         """Given an IRI for which a prefix is known (for example
         "http://www.w3.org/2000/01/rdf-schema#label") this method returns a
-        CURIE (for example "rdfs:label"), if no prefix is known the original
-        IRI is returned."""
-        return to_curie(iri, self)
+        CURIE (for example "rdfs:label"). If no prefix is known the original
+        IRI is returned. If a base is provided and IRI starts with that base,
+        the base is stripped out and replaced with '#'."""
+        curie = to_curie(iri, self)
+        if base and curie.startswith(base):
+            return curie.replace(base, "#")
```

Comment:
This still isn't quite right; for example it's valid to have a base of `https://example.com/` and that would serialize `https://example.com/foo` to `<foo>`.

Reply:
Ah, okay, it makes sense to adjust for all valid bases and not add an unneeded "#" if not applicable. 👍